tests/codegen/issue-73825-gvn-const-local-array.rs

```diff
@@ -0,0 +1,25 @@
+// issue: <https://github.com/rust-lang/rust/issues/73825>
+//@ compile-flags: -C opt-level=1
+#![crate_type = "lib"]
+
+// CHECK-LABEL: @foo
+// CHECK-NEXT: {{.*}}:
+// CHECK-NEXT: and
+// CHECK-NEXT: getelementptr inbounds
+// CHECK-NEXT: load i32
+// CHECK-NEXT: ret i32
```

Comment:
Could you make the checks more explicit? Reading this, I have no idea what the generated IR looks like.